giskard/llm/loaders/prompt_injections.py

```diff
@@ -0,0 +1,97 @@
+import ast
+from typing import Optional
+import pandas as pd
+
+from ...datasets.base import Dataset
+from ..evaluators.string_matcher import StringMatcherConfig
+
+INJECTION_DATA_URL = "https://raw.githubusercontent.com/Giskard-AI/prompt-injections/main/prompt_injections.csv"
+GISKARD_META_URL = "https://raw.githubusercontent.com/Giskard-AI/prompt-injections/main/giskard_meta_data.csv"
+
+
+def _check_matching_dfs_len(df1, df2):
+    if len(df1) != len(df2):
+        raise ValueError(
+            f"{__name__}: {INJECTION_DATA_URL} and {GISKARD_META_URL} should "
+            "have the same length and should be a one-to-one mapping of each other."
+        )
+
+
+def _check_meta_df_requirements(df):
+    if "expected_strings" not in df.columns:
+        raise ValueError(f"{__name__}: expected_strings are needed for the evaluation.")
+
+    if df.expected_strings.isnull().values.any():
+        raise ValueError(f"{__name__}: expected_strings column cannot have any NaN values.")
+    df.expected_strings = df.expected_strings.apply(ast.literal_eval)
+
+
+class PromptInjectionDataLoader:
+    def __init__(
+        self,
+        num_samples: Optional[int] = None,
+    ):
+        self.num_samples = num_samples
+        self._df = None
+
+    def load_dataset_from_group(self, features, group) -> Dataset:
+        prompts = self.prompts_from_group(group)
+        prompts = pd.DataFrame({feature: prompts for feature in features}, index=prompts.index)
+        return Dataset(
+            df=prompts,
+            name="Injection Prompts",
+            target=None,
+            cat_columns=None,
+            validation=False,
+        )
+
+    @property
+    def df(self):
+        if self._df is None:
+            prompt_injections_df = pd.read_csv(INJECTION_DATA_URL, index_col=["index"])
+            meta_df = pd.read_csv(GISKARD_META_URL, index_col=["index"])
+            _check_matching_dfs_len(meta_df, prompt_injections_df)
+            _check_meta_df_requirements(meta_df)
+            self._df = prompt_injections_df.join(meta_df)
+
+            if self.num_samples is not None:
+                self._df = self._df.sample(self.num_samples)
+                self._df.reset_index(inplace=True, drop=True)
+
+        return self._df
+
+    @property
+    def names(self):
+        return self.df.name.tolist()
+
+    @property
+    def groups(self):
+        return list(set(self.df.group_mapping.tolist()))
```

Comment:
pandas style could be more concise `self.df.group_mapping.unique().tolist()`

Reply:
done